qiskit/pulse/schedule.py

```diff
@@ -186,6 +188,87 @@ def flatten(self) -> 'ScheduleComponent':
         """Return a new schedule which is the flattened schedule contained all `instructions`."""
         return ops.flatten(self)
 
+    def filter(self, *filter_funcs: List[Callable],
+               channels: Iterable[int] = [],
+               instruction_types: Iterable[abc.ABCMeta] = [],
+               intervals: Iterable[Tuple[int, int]] = []) -> 'Schedule':
+        """
+        Return a new Schedule with only the instructions which pass though the provided filters.
+        Custom filters may be provided. If a list of channel indices is provided, only the
+        instructions that involve that channel (and maybe also others) will be included in the new
+        schedule. Similarly for instruction_types, only the instructions which are instances of the
+        provided types will be included. For intervals, instructions will be retained if their
+        timeslots are all wholly contained within *any* of the given intervals.
+
+        If no arguments are provided, this schedule is returned.
+
+        Args:
+            filter_funcs: A list of Callables which take a (int, ScheduleComponent) tuple and
+                          return a bool
+            channels: Channel indices, e.g. {0, 1, 2}
+            instruction_types: For example, [PulseInstruction, AcquireInstruction]
+            intervals: Time intervals to keep, e.g. [(0, 5), (6, 10)]
+        """
+        def only_channels(channels: Set[int]) -> Callable:
```

Comment:
I think it is desired for the `channels` to be iterable containing type `Channel`. That way we can specify say only `DriveChannel(0)` or `AcquireChannel(1)`.